src/js/pensions/config/form.js

```diff
@@ -245,7 +265,28 @@ const formConfig = {
                   'Date entered service must be before date left service'
                 )
               }
-            }
+            },
+            'view:wartimeWarning': (() => {
+              const rangeIncludesWartime = createSelector(
+                form => form.servicePeriods,
+                (periods) => {
+                  return (periods || []).some(period => {
```

Comment:
I feel like this might be clearer if you separated out the full date checks from the call to check the service period. The way I understand the logic is that we want to filter out any periods with incomplete dates (since those are likely in progress). Then, if there's at least one period with full dates, we want to show the warning if none of those periods are in wartime.

```js
const completePeriods = (periods || []).filter(period => {
  const isFullDate = /^\d{4}-\d{2}-\d{2}$/;
  return period.activeServiceDateRange &&
    isFullDate.test(period.activeServiceDateRange.to) &&
    isFullDate.test(period.activeServiceDateRange.from);
});

if (!periods.length) {
  return true;
}

return completePeriods.some(period => servedDuringWartime(period.activeServiceDateRange);
```